drivers/virtio/net/virtio_net.c

```diff
@@ -263,8 +265,7 @@ static void virtio_netdev_xmit_free(struct uk_netdev_tx_queue *txq)
 #define RX_FILLUP_BATCHLEN 64
 
 static int virtio_netdev_rx_fillup(struct uk_netdev_rx_queue *rxq,
-				   __u16 nb_desc,
-				   int notify)
+				   __u16 nb_desc, int notify, __u8 is_modern)
```

Comment:
I think it would me more useful to have the size of the virtio header as an argument, instead of doing so many if-else checks based on the `is_modern` flag.

Reply:
Yes, I don't like it very much either. I came up with what I believe to be a cleaner approach where we pass the device as a parameter, and use a macro to derive the header size. It also adds the `numb_buffers` field into the header and allows to transparently add the fields that depend on `VIRTIO_NET_F_HASH_REPORT` later.